docs/Project-Configuration/Project-Configuration-Files.md

```diff
@@ -1515,6 +1515,10 @@ The following constraints apply:
 -   Task groups cannot have tags applied to them and cannot be selected by
     tags.
 
+#### The following constraints apply to single host task groups:
+
+-   If tasks in a single host task groups have dependencies on another task outside the group that task should be the first task in the group. Only the first task in the group should have dependencies outside of the group to avoid the host being blocked and hitting an idle timeout while waiting for that external dependency to complete. 
```

Comment:
nit: might be clearer to say something like `If tasks in a single host task groups have dependencies on another task outside the group, only the first task in the task group should list those dependencies. If a task in the group other than the first one have dependencies outside of the group, the task can be blocked waiting for external dependencies to complete and result in the host being terminated for idleness.`